### PR TITLE
Allow to disable checksums on read

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add an option to disable validation of checksums on reading. Should never be used in production. Please do not expect any benefits in disabling it. It may only be used for experiments and benchmarks. The setting only applicable for tables of MergeTree family. Checksums are always validated for other table engines and when receiving data over network. In my observations there is no performance difference or it is less than 0.5%.

### CODE REVIEW

File: src/Core/Settings.h [hunk 1]
```diff
@@ -139,6 +139,7 @@ class IColumn;
     \
     M(UInt64, min_bytes_to_use_direct_io, 0, "The minimum number of bytes for reading the data with O_DIRECT option during SELECT queries execution. 0 - disabled.", 0) \
     M(UInt64, min_bytes_to_use_mmap_io, 0, "The minimum number of bytes for reading the data with mmap option during SELECT queries execution. 0 - disabled.", 0) \
+    M(Bool, checksum_on_read, true, "Validate checksums on reading. It is enabled by default and should be always enabled in production. Please do not expect any benefits in disabling this setting. It may only be used for experiments and benchmarks. The setting only applicable for tables of MergeTree family. Checksums are always validated for other table engines and when receiving data over network.", 0) \
```
Comment (alesapin) [inline src/Core/Settings.h:142]: Maybe make this name longer (and less convenient)? `merge_tree_disable_checksum_on_read`?

Comment (alexey-milovidov) [reply src/Core/Settings.h:142]: We have settings like `min_bytes_to_use_mmap_io` that are also applied only for MergeTree but they can be extended for Log-engines. I wish to treat `checksum_on_read` in the same way.

**Comment by alexey-milovidov** [comment]
In my observations there is no performance difference or it is less than 0.5%.

**Comment by alexey-milovidov** [comment]
I don't think it needs any tests.

**Comment by alexey-milovidov** [comment]
#19754

**Comment by alexey-milovidov** [comment]
#19755

**Pull request merged**